include/gba/isagbprint.h

```diff
@@ -1,50 +1,58 @@
 #ifndef GUARD_GBA_ISAGBPRINT_H
 #define GUARD_GBA_ISAGBPRINT_H
 
+#include "gba/types.h"
+
 #ifdef NDEBUG
+#define DebugPrintf(pBuf, ...)
+#define MgbaOpen()
+#define MgbaClose()
+#define AGB_ASSERT(exp)
+#define AGB_WARNING(exp)
 #define AGBPrintInit()
-#define AGBPutc(cChr)
-#define AGBPrint(pBuf)
-#define AGBPrintf(pBuf, ...)
-#define AGBPrintFlush1Block()
-#define AGBPrintFlush()
-#define AGBAssert(pFile, nLine, pExpression, nStopProgram)
 #else
-void AGBPrintInit(void);
-void AGBPutc(const char cChr);
-void AGBPrint(const char *pBuf);
+#if (LOG_HANDLER == LOG_HANDLER_MGBA_PRINT)
+bool32 MgbaOpen(void);
+void MgbaClose(void);
+void MgbaPrintf(const char *pBuf, ...);
+void MgbaAssert(const char *pFile, s32 nLine, const char *pExpression, bool32 nStopProgram);
+#define DebugPrintf(pBuf, ...) MgbaPrintf(pBuf, __VA_ARGS__)
+#define AGB_ASSERT(exp) (exp) ? ((void*)0) : MgbaAssert(__FILE__, __LINE__, #exp, 1)
+#define AGB_WARNING(exp) (exp) ? ((void*)0) : MgbaAssert(__FILE__, __LINE__, #exp, 0)
```

Comment:
I think these should use TRUE or FALSE for `nStopProgram` uses.

Reply:
22329d3318cde3f07ed5401226c0df2b5d951957